WordPress/WordPressTest/EUUSCompliance/CompliancePopoverViewModelTests.swift

```diff
@@ -0,0 +1,87 @@
+import XCTest
+@testable import WordPress
+
+final class CompliancePopoverViewModelTests: CoreDataTestCase {
+    let testDefaults = UserDefaults(suiteName: "compliance-popover-view-model-tests")
+
+    override func setUp() {
+        super.setUp()
+        testDefaults?.removeObject(forKey: UserDefaults.didShowCompliancePopupKey)
+
+        let windowManager = WindowManager(window: UIWindow())
+        WordPressAuthenticationManager(
+            windowManager: windowManager,
+            remoteFeaturesStore: RemoteFeatureFlagStore()
+        ).initializeWordPressAuthenticator()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        testDefaults?.removeObject(forKey: UserDefaults.didShowCompliancePopupKey)
+    }
+
+    func testDidTapSettingsUpdatesDefaults() {
+        guard let defaults = try? XCTUnwrap(testDefaults) else {
+            return
+        }
+        let sut = CompliancePopoverViewModel(defaults: defaults, contextManager: contextManager)
+        sut.didTapSettings()
+        XCTAssert(defaults.didShowCompliancePopup)
+    }
+
+    func testDidTapSettingsInvokesCoordinatorNavigation() {
+        guard let defaults = try? XCTUnwrap(testDefaults) else {
+            return
+        }
+        let mockCoordinator = MockCompliancePopoverCoordinator()
+        let sut = CompliancePopoverViewModel(defaults: defaults, contextManager: contextManager)
+        sut.coordinator = mockCoordinator
+        sut.didTapSettings()
+        XCTAssertEqual(mockCoordinator.navigateToSettingsCallCount, 1)
+    }
+
+    func testDidTapSaveInvokesDismissWhenAccountIDExists() {
+        guard let defaults = try? XCTUnwrap(testDefaults) else {
+            return
+        }
+        let mockCoordinator = MockCompliancePopoverCoordinator()
+        let sut = CompliancePopoverViewModel(
+            defaults: defaults,
+            contextManager: makeCoreDataStack()
+        )
+        sut.coordinator = mockCoordinator
+        sut.didTapSave()
+        XCTAssertEqual(mockCoordinator.dismissCallCount, 1)
+        XCTAssert(defaults.didShowCompliancePopup)
+    }
+
+    private func makeCoreDataStack() -> ContextManager {
+        let contextManager = ContextManager.forTesting()
+        let account = AccountBuilder(contextManager)
+            .with(id: 1229)
+            .with(username: "foobar")
+            .with(email: "foo@automattic.com")
+            .with(authToken: "9384rj398t34j98")
+            .build()
+        UserSettings.defaultDotComUUID = account.uuid
+        return contextManager
+    }
```

Comment:
`CoreDataTestCase` already offers a `ContextManager` instance:

https://github.com/wordpress-mobile/WordPress-iOS/blob/71ac3120754f2b30890f9233006e2bc1c0f7817d/WordPress/WordPressTest/CoreDataTestCase.swift#L6-L10

This one off logic could be inlined in the one test using it, which has also IMHO the benefit of making what "when account id exists" means, instead of burying that setup step in a function whose name doesn't hint to it.

```suggestion
    func testDidTapSaveInvokesDismissWhenAccountIDExists() {
        let defaults = try XCTUnwrap(testDefaults)
        
        let account = AccountBuilder(contextManager)
            .with(id: 1229)
            .with(username: "foobar")
            .with(email: "foo@automattic.com")
            .with(authToken: "9384rj398t34j98")
            .build()
        UserSettings.defaultDotComUUID = account.uuid

        let mockCoordinator = MockCompliancePopoverCoordinator()
        let sut = CompliancePopoverViewModel(
            defaults: defaults,
            contextManager: contextManager
        )
        sut.coordinator = mockCoordinator
        
        sut.didTapSave()
        
        XCTAssertEqual(mockCoordinator.dismissCallCount, 1)
        XCTAssert(defaults.didShowCompliancePopup)
    }
```

Reply:
Something I love about tests is how they surface API ergonomics.

**This is unrelated to this PR**, but I want to make the comment here so it's closer to the context.

I found it odd that we setup a dedicated `UserDefaults` instance (which I think is good) but then manipulate it not directly but via static methods on `UserSettings`. The only way for someone to know they're supposed to do that is by reading some documentation or by having someone else telling them about it. And granted, it's totally acceptable that a complex project like this one would come with knowledge that needs to be transferred or documented. But wouldn't it be better if the type system or the software design could help us with that?

For example, given `UserSettings` manipulates _only_ `UserDefaults`, we could have the logic implemented on `UserDefaults` directly.

Or, and IMHO better, we could separate the collection of user settings into a `struct` (`UserSettings`) and abstract their storage behind a `protocol` (`UserSettingsStoring`, perhaps?). This would allow us to 

1. Not having to worry about settings storage in most case, because we could simply pass a `UserSettings` instance
2. Decouple the low level implementation details of how settings are stored from the API consumers call when updating them, which in turns would allow us to use test double in most unit tests always working in memory and bypassing the need created dedicated `UserDefaults` or having to store and restore the state on `.defaults`

Reply:
I definitely got confused when I was trying to set the account value for the `ContextManager`. It seems totally unlinked that when we do `UserSettings.defaultDotComUUID = account.uuid` ContextManager is affected by this change. It is probably why I added a redundant ContextManager. Although I don't think there's anything wrong with extracting account creation to a function. I think it is even better to do it that way to keep the test function lest bloated as it wouldn't care how account is instantiated, furthermore another test function could need the same thing. So if that's fine with you, I'll use the `ContextManager` from the `CoreDataTestCase` and leave the function separate.

Reply:
> Although I don't think there's anything wrong with extracting account creation to a function. I think it is even better to do it that way to keep the test function lest bloated as it wouldn't care how account is instantiated [...]. So if that's fine with you, I'll use the `ContextManager` from the `CoreDataTestCase` and leave the function separate.

Sure. In that case, though, how would you feel about renaming it from `makeCoreDataStack`, which has no information about the important fact that an account is needed, to something like `configureAccountInContextManager`, or `addAccountToCoreManager`, or `addAccount(to: contextManager)`? We might not care more about how the account is created, but we do care about the fact that there is an account.

> , furthermore another test function could need the same thing.

Ah, fair enough 😄  Although, one could say YAGNI.